src/main/kotlin/io/libp2p/security/noise/NoiseXXSecureChannel.kt

```diff
@@ -83,63 +94,10 @@ class NoiseXXSecureChannel(val localKey: PrivKey, val localDHState: DHState, val
             logger.debug("Starting handshake")
         }
 
-        override fun channelRead0(ctx: ChannelHandlerContext?, msg: ByteBuf?) {
-            channelRead(ctx!!, msg as Any)
-            super.channelRead(ctx, msg)
-        }
-
-        override fun channelRegistered(ctx: ChannelHandlerContext?) {
-            super.channelRegistered(ctx)
-
-            if (activated) {
-                return
-            }
-            logger.debug("Registration starting")
-            activated = true
-
-            if (role == HandshakeState.INITIATOR) {
-                val msgBuffer = ByteArray(65535)
-
-                // TODO : include data fields into protobuf struct to match spec
-                // alice needs to put signed peer id public key into message
-                val signed = localKey.sign(localKey.publicKey().bytes())
-
-                // generate an appropriate protobuf element
-                val bs = Spipe.Exchange.newBuilder().setEpubkey(ByteString.copyFrom(localKey.publicKey().bytes()))
-                    .setSignature(ByteString.copyFrom(signed)).build()
-
-                // create the message
-                // also create assign the signed payload
-                val msgLength = handshakestate.writeMessage(msgBuffer, 0, bs.toByteArray(), 0, bs.toByteArray().size)
-
-                // put the message frame which also contains the payload onto the wire
-                val writeAndFlush = ctx?.writeAndFlush(msgBuffer.copyOfRange(0, msgLength))
-                writeAndFlush?.await()
-            }
-            logger.debug("Registration complete")
-        }
-
-        override fun channelActive(ctx: ChannelHandlerContext?) {
-            logger.debug("Activation starting")
-            super.channelActive(ctx)
-            channelRegistered(ctx)
-            logger.debug("Activation complete")
-        }
-
-        private var activated = false
-        private var flagRemoteVerified = false
-        private var flagRemoteVerifiedPassed = false
-        private var aliceSplit: CipherState? = null
-        private var bobSplit: CipherState? = null
-        private var cipherStatePair: CipherStatePair? = null
-
-        override fun channelRead(ctx: ChannelHandlerContext, msg1: Any) {
+        override fun channelRead0(ctx1: ChannelHandlerContext?, msg1: ByteBuf?) {
```

Comment:
You can safely use non-null types for params here : 
```kt
channelRead0(ctx: ChannelHandlerContext, msg: ByteBuf)
```

Reply:
done